### PR TITLE
To Do list: add & remove

### PR DESCRIPTION
**Pull request by Stephen-Adom**
# To Do list: add & remove
In this pull request, the following tasks were implemented:
- Removed all hardcoded items from the tasks array.
- Implemented a function for adding a new task
- Implemented a function for adding a new task
- Implemented a function for editing task descriptions
- Reset the index when todo items are deleted
- Save todos to localstorage

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -1,10 +1,45 @@
+/* eslint-disable import/no-extraneous-dependencies */
 import './styles.css';
-import arrowDown from './assets/bend-arrow-right-svgrepo-com.svg';
+import { debounceTime, fromEvent } from 'rxjs';
```
Comment (AmaduKamara) [inline src/index.js]: - [ ] Let's try to not disable Linter checks in our codebase as they help us write better, clean, and professional code. Disabling Lintger checks defies its purpose. Kindly fix all linter issues and resubmit your project for approval 👍🏽

File: src/modules/add_remove_task.js [hunk 2]
```diff
@@ -0,0 +1,46 @@
+// eslint-disable-next-line import/no-extraneous-dependencies
+import { fromEvent, debounceTime } from 'rxjs';
+import Todo from './todo.model.js';
```
Comment (AmaduKamara) [inline src/modules/add_remove_task.js]: - [ ] Let's try to not disable Linter checks in our codebase as they help us write better, clean, and professional code. Disabling Lintger checks defies its purpose. Kindly fix all linter issues and resubmit your project for approval 👍🏽

File: src/modules/add_remove_task.js [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+
+// eslint-disable-next-line import/no-mutable-exports
+let TodoList = [];
```
Comment (AmaduKamara) [inline src/modules/add_remove_task.js]: - [ ] Let's try to not disable Linter checks in our codebase as they help us write better, clean, and professional code. Disabling Lintger checks defies its purpose. Kindly fix all linter issues and resubmit your project for approval 👍🏽

**Review by AmaduKamara — CHANGES_REQUESTED** [review]
# Status: Required Changes ♻️ ♻️ ♻️

Hi @Stephen-Adom,

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights

✅ You used Gitflow correctly 👍🏽
✅ Linter checks passed 👍🏽
✅ Add, update, and remove todo features work fine as expected 👍🏽
✅ The app is nicely styled 👍🏽
✅ Good commit messages and good descriptive PR 👍🏽

## Required Changes ♻️
_Check the comments under the review._

- [ ] We're not allowed to disable Linters in our codebase. Kindly fix them 👍🏽

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
```

File: src/modules/add_remove_task.js [hunk 4]
```diff
@@ -0,0 +1,45 @@
+import { fromEvent, debounceTime } from 'rxjs';
+import Todo from './todo.model.js';
+
+// eslint-disable-next-line import/no-mutable-exports
```
Comment (OLIPLICHE) [inline src/modules/add_remove_task.js]: - [ ] I would advise you to fix this linter manually as said in the previous review. You can use [`StackOverflow`](https://stackoverflow.com/questions/67713204/eslint-disable-default-export) website for research on how to fix the challenge you are facing.

**Review by OLIPLICHE — CHANGES_REQUESTED** [review]
Hi @Stephen-Adom , 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.


 ## Highlights
- README file is descriptive 👏
- Commit messages are meaningful 👏 👏

 ## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by KayLemba — APPROVED** [review]
[[# STATUS: APPROVED :green_circle: :clap: 

Hello @Stephen-Adom     , 
Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

HIGHLIGHTS:

- [x]  Your repo is professional 
- [x] Descriptive P.R
- [x] Proffessional README









 








Now you can merge this branch and move on :1st_place_medal: 
Happy Coding cheers 🥇 💯 



well-1

Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better.

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._]]

**Pull request merged**